File: Test2-8.py
def first():
    a = 10
    
    def second():
        nonlocal a
        b = a
        c = 50
        print('Variable b references a from function first() so b = ',b)
        try:
            print('variable d: ',d)
        except:
            print('accessing the variable d from non-local function... ')
        
        def third():
            nonlocal a
            nonlocal c
            i = a
            j = c
            print('Variable i references a from function first() so i = ',i)
            print('Variable j references c from function second() so j = ',j)
        third()    
    second()
    d = 20
    print(' In the function first(), variable d =',d)
first()

File: Test2-8.js
//Test 2 Q8
const first = () => {
  let a = 100;
  console.log(`In the first() function, a = ${a}`);

  const second = () => {
    a = 200;
    console.log(`In the second() function, a = ${a}`);
    try {
      console.log(`In the second() function, b = ${b}`);
    } catch (err) {
      console.log("Variable b cannot be accessed by the child function");
    }
    const third = (b) => {
      a = 300;
      console.log(`In the third() function, a = ${a}`);
    };
    third();
  };
  second();
  const b = 0;

  console.log(`After these function calls a = ${a}`);
  console.log(`Variable b after these function calls = ${b}`);
};
first();
